module-vfs/drivers/src/purefs/fs/filesystem_vfat.cpp

```diff
@@ -139,6 +139,42 @@ namespace purefs::fs::drivers
             st.st_mtime  = 0;
             st.st_ctime  = 0;
         }
+
+        int internal_stat_rootdir(const char* translated_path, bool ro, struct stat* entry)
+        {
+            FATFS *fs;
+            DWORD bfree;
+            FRESULT res = f_getfree(translated_path, &bfree, &fs);
+            if (res != FR_OK)
+            {
+                return -EIO;
+            }
+            entry->st_dev = 0;
+            entry->st_ino = 0;
+            entry->st_mode = (S_IFDIR | S_IXUSR | S_IXGRP | S_IXOTH | S_IRUSR | S_IRGRP | S_IROTH);
+            if(!ro) {
+                entry->st_mode |= S_IWUSR | S_IWGRP | S_IWOTH;
+            }
+            entry->st_nlink = 1;
+            entry->st_uid = 0;
+            entry->st_gid = 0;
+            entry->st_rdev = 0;
+            entry->st_size = 0;
+            // TODO: Block FF_MIN_SS != FF_MAX_SS
+#if FF_MAX_SS != FF_MIN_SS
+            entry->st_blksize = fatfs->ssize;
+#else
+            entry->st_blksize = FF_MIN_SS;
+#endif
+            // TODO: Time is currently not supported
```

Comment:
instead of todo in the code, consider new jira case

Reply:
fixed